build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/tscTask.ts

```diff
@@ -612,7 +611,7 @@ export class TscMultiTask extends LeafWithDoneFileTask {
 				return { name, hash };
 			});
 
-			const buildInfo = readFileSync(tsbuildinfoPath).toString();
+			const buildInfo = (await readFile(tsbuildinfoPath)).toString();
```

Comment:
I thought linter didn't like `(await Foo).access` pattern.

Reply:
build-tools lint rules are old - known issue and one of the many things we need to clean up. The stated reason for the lint rule is: "When accessing a member from an await expression, the await expression has to be parenthesized, **which is not readable.**"

I... disagree. This seems plenty readable to me and moreso than splitting it over two lines - though I imagine _some_ cases would be more confusing.

Is there another reason to avoid this pattern?

Reply:
The only reason I know is for debuggability. It is hard to step through the await and then check the next steps (which may be more complicated than a simple value read. There isn't a real penalty for capturing the promise result; so, I've never fought the rule.